File: backend/app/models/title.py
from typing import Optional
from sqlalchemy import UniqueConstraint
from sqlmodel import Field, SQLModel

class Title(SQLModel, table=True):
    """
    Stores movies/TV shows you can rank and recommend.
    """

    id: Optional[int] = Field(default=None, primary_key=True)

    __table_args__ = (
        UniqueConstraint("tmdb_id", "type", name="uq_title_tmdb_type"),
    )

    tmdb_id: Optional[int] = Field(default=None)

    name: str = Field(nullable=False)

    # "movie" | "tv" stored as string for MVP
    type: str = Field(nullable=False)

    year: Optional[int] = Field(default=None)

    poster_url: Optional[str] = Field(default=None)

    # MVP: comma-separated genres string
    genres: Optional[str] = Field(default=None)

    overview: Optional[str] = Field(default=None)

    runtime_minutes: Optional[int] = Field(default=None)

File: frontend/src/types/title.ts
export type Title = {
  id: number;
  name: string;
  type: string;
  year: number | null;
  poster_url: string | null;
  genres: string | null;
  tmdb_id: number | null;
  overview?: string | null;
  runtime_minutes?: number | null;
};

export type ExternalTitle = {
  id: string;
  tmdb_id: number;
  type: "movie" | "tv";
  name: string;
  year: number | null;
  poster_url: string | null;
  overview: string | null;
  genre_ids: number[];
  genres: string[];
  runtime_minutes: number | null;
  local_title_id: number | null;
  average_score: number | null;
  rating_count: number;
};

export type SelectableTitle = Title | ExternalTitle;

export function isExternalTitle(title: SelectableTitle): title is ExternalTitle {
  return typeof title.id === "string";
}
